database_refresh/README.md

```diff
@@ -1,88 +1,98 @@
+- `03-prometheus-alert.yaml` Define alert which will tell us if the snapshot job failed, alert sent to `#dps_alerts`
 
-If setting this up for the first time ***use extreme caution*** - this job connects to production databases.
+Also:
 
-***Ensure that all environment variables are named and setup correctly.***
+- `rds-snapshot-job.yaml` Can be used to create one off snapshots on an adhoc basis.
 
-Also check that the rule defined in `03-prometheus-alert.yaml` has the correct namespace specified, update the the rule if needed.
+### Prerequisite:
 
-### Prerequisite:
+For this job to work the AWS IAM creds, exported by the terraform module, need to be exported into the namespace as secrets.  These creds have limited permissions which allow for managing snapshots.
 
-For this job to work the preprod database credentials need to be available production namespace. This is achieved by having terraform export the preprod credentials, which is an output of the terraform rds module, to a secret in the production namespace.  See this example:
+See module outputs:
+- `access_key_id`
+- `secret_access_key`
 
-<https://github.com/ministryofjustice/cloud-platform-environments/blob/master/namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-categorisation-preprod/resources/rds.tf#L40>
+Source code here: <https://github.com/ministryofjustice/cloud-platform-terraform-rds-instance/>
+
+E.g. <https://github.com/ministryofjustice/cloud-platform-environments/blob/master/namespaces/live-1.cloud-platform.service.justice.gov.uk/offender-categorisation-preprod/resources/rds.tf#L52>
+
+Also required module output is `rds_instance_address` which is mapped to env var `DB_INSTANCE`.
 
 ### Overview
 
-The refresh job performs a `pg_dump` using the existing production credentials, already setup in in the production namespace of the application.  The job then uses the preprod credentials (see prerequisite) for the `pg_restore`.
+The snapshot job runs on a schedule determined in `02-cronjob.yaml`.  Two variables can be configured in the `01-configmap-db-snapshot-script.yaml`:
 
-In sync with the main oracle database refreshes this cronjob runs every other Sunday (odd weeks).  Slight caveat is that is it difficult to define a cron schedule that runs every other week.  So this cron is scheduled for every week however the executed script checks if the week is an odd or even number and only executes the refresh on odd numbered weeks.
+- `SNAPSHOT_PREFIX`, prefix for snapshot name, snapshot is postfixed with epoch timestamp.
+- `RETENTION`, the number of snapshots to keep.
 
 ### Installation of cronjob
 
 ```bash
-kubectl -n offender-categorisation-prod apply -f 01-configmap-refresh-script.yaml
-kubectl -n offender-categorisation-prod apply -f 02-cronjob.yaml
-kubectl -n offender-categorisation-prod apply -f 03-prometheus-alert.yaml
+kubectl -n [namespace] apply -f 01-configmap-db-snapshot-script.yaml
```

Comment:
leaving [namespace] placeholder as in keyworker example - also can be applied to dev/preprod/prod namespaces